conans/test/integration/command/create_test.py

```diff
@@ -409,3 +409,140 @@ def test_create_build_missing():
+    user.second:value+=["other value"]
+    [buildenv]
+    VAR1=myvalue1
+    """)
+    profile_host = textwrap.dedent("""\
+    [settings]
+    arch=x86
+    build_type=Debug
+    compiler=gcc
+    compiler.libcxx=libstdc++
+    compiler.version=12
+    os=Linux
+    """)
+    conanfile = textwrap.dedent("""
+        from conan import ConanFile
+
+        class MyTest(ConanFile):
+            name = "pkg"
+            version = "0.2"
+            settings = "build_type", "compiler"
+            author = "John Doe"
+            license = "MIT"
+            url = "https://foo.bar.baz"
+            homepage = "https://foo.bar.site"
+            topics = "foo", "bar", "qux"
+            provides = "libjpeg", "libjpg"
+            deprecated = "other-pkg"
+            options = {"shared": [True, False], "fPIC": [True, False]}
+            default_options = {"shared": False, "fPIC": True}
+        """)
+    client.save({"conanfile.py": conanfile,
+                 "host": profile_host, "build": profile_build})
+    client.run("create . -pr:h host -pr:b build")
+    client.save({"conanfile.py": GenConanfile().with_name("hello").with_version("0.1")
+                                               .with_require("pkg/0.2"),
+                 "host": profile_host, "build": profile_build}, clean_first=True)
+    client.run("create . -f json -pr:h host -pr:b build")
+    info = json.loads(client.stdout)
+    profile_build = {'build_env': 'VAR1=myvalue1\n',
+                     'conf': {'user.second:value': ['my value', 'other value'],
+                              'user.first:value': '"my value"'},
+                     'options': {},
+                     'package_settings': {},
+                     'settings': {'arch': 'x86_64',
+                                  'build_type': 'Release',
+                                  'compiler': 'gcc',
+                                  'compiler.libcxx': 'libstdc++',
+                                  'compiler.version': '12',
+                                  'os': 'Linux'},
+                     'tool_requires': {}}
+    assert profile_build == info["profile_build"]
+
+    profile_host = {'build_env': '',
+                    'conf': {},
+                    'options': {},
+                    'package_settings': {},
+                    'settings': {'arch': 'x86',
+                                 'build_type': 'Debug',
+                                 'compiler': 'gcc',
+                                 'compiler.libcxx': 'libstdc++',
+                                 'compiler.version': '12',
+                                 'os': 'Linux'},
+                    'tool_requires': {}}
+    assert profile_host == info["profile_host"]
+    graph_info = info["graph"]
+    hello_pkg_ref = 'hello/0.1#18d5440ae45afc4c36139a160ac071c7'
+    pkg_pkg_ref = 'pkg/0.2#db78b8d06a78af5c3ac56706f133098d'
+    assert graph_info["conanfile"]["recipe"] == "Virtual"
+    assert graph_info["conanfile"]["package_id"] is None
```

Comment:
why None? I don't get why `package_Id` of conanfile cannot be computed. same for PREV.

Reply:
Because this is the consumer information, purely, your ` conanfile.py` one. Then, that `package_id` has to be `None`, and, if you read below, you'll see `assert graph_info[hello_pkg_ref]["package_id"] == "8eba237c0fb239fcb7daa47979ab99258eaaa7d1"` which is the created package for that conanfile.